server/src/main/java/io/spine/server/entity/storage/ColumnReader.java

```diff
@@ -91,27 +93,34 @@ static ColumnReader forClass(Class<? extends Entity> entityClass) {
         }
 
         Collection<EntityColumn> entityColumns = newLinkedList();
-        for (PropertyDescriptor property : entityDescriptor.getPropertyDescriptors()) {
-            Method getter = property.getReadMethod();
-            boolean isEntityColumn = getAnnotatedVersion(getter).isPresent();
-            if (isEntityColumn) {
-                EntityColumn column = EntityColumn.from(getter);
-                entityColumns.add(column);
-            }
-        }
+        PropertyDescriptor[] propertyDescriptors = entityDescriptor.getPropertyDescriptors();
+        Arrays.stream(propertyDescriptors)
+              .filter(ColumnReader::hasAccessor)
+              .filter(ColumnReader::accessorIsAnnotated)
+              .map(PropertyDescriptor::getReadMethod)
+              .forEach(readMethod -> entityColumns.add(EntityColumn.from(readMethod)));
```

Comment:
Please use a combination of `map` and `collect` instead. `forEach` must be stateless. In practice, in order to be honest with the functional style coding, `forEach` is almost never applicable.

Reply:
Also, you may extract the method from the descriptor as the first step, filter out nulls and not annotated methods and collect them at once. This is also a performance improvement, since `PropertyDescriptor.getReadMethod()` looks quite complex and not necessarily fast.